lnwallet/channel.go

```diff
@@ -4800,7 +4820,7 @@ func (lc *LightningChannel) resignMusigCommit(commitTx *wire.MsgTx,
 // previous commitment txn. This allows the link to clear its mailbox of those
 // circuits in case they are still in memory, and ensure the switch's circuit
 // map has been updated by deleting the closed circuits.
-func (lc *LightningChannel) ProcessChanSyncMsg(
+func (lc *LightningChannel) ProcessChanSyncMsg(quit <-chan struct{},
```

Comment:
Can we make this a context instead? That way the context from the main coordination goroutine, or RPC calls can pass the context in, and even create a timeout, etc.

Reply:
That sounds reasonable, but I don't see a context from `ChannelLink` that we could actually inherit; only the quit channel.

We could build a Context from that quit channel with some background goroutine that cancels a context if a channel is closed:

```go
func cancelOnQuit(cancel context.CancelFunc, quit <-chan struct{}) {
	for {
		select {
		case <-quit:
			cancel()
			return
		default:
		}
	}
```

That does feel a bit hacky IMO, and I'm not sure if we can neatly 'combine' two contexts. We could pass both the 'quit' channel, and a context.

Reply:
Alternatively, we can pass the new quit channel/context in via the main constructor. It has functional options set. This way, we don't need to change all call sites with the new input argument. Then in the link, we signal the channel state machine to exit. 

If we go this path, then we'll need to verify some assumptions re state/pointer sharing amongst the: switch, cnct, and peer.

Reply:
Also you can get a channel to listen on for done like via the `Done` method: https://pkg.go.dev/context#Context. 

> That sounds reasonable, but I don't see a context from ChannelLink that we could actually inherit; only the quit channel.

Yeah so along the way we'd need to add a context to the link. 

Re my suggestion above: is it correct that we only want things to bail out once the peer/link does? Or does a custom chans caller need to be able to cancel things earlier?

Reply:
> Re my suggestion above: is it correct that we only want things to bail out once the peer/link does? 

I think this is correct / the problem this PR should be fixing.

> Or does a custom chans caller need to be able to cancel things earlier?

Not sure, but it does add another spot for a blocking read where we do want to listen on this quit signal.